services/federator/test/integration/Test/Federator/InwardSpec.hs

```diff
@@ -84,6 +84,24 @@ spec env =
         err <- asInwardErrorUnsafe <$> inwardBrigCall "federation/this-endpoint-does-not-exist" (encode Aeson.emptyObject)
         expectErr IInvalidEndpoint err
 
+    it "should only not accept invalid paths" $
+      runTestFederator env $ do
+        let o = object ["name" .= ("fakeNewUser" :: Text)]
+        err <- asInwardErrorUnsafe <$> inwardBrigCall "http://localhost:8080/i/users" (encode o)
+        expectErr IInvalidEndpoint err
+
+    it "should only accept /federation/ paths" $
+      runTestFederator env $ do
+        let o = object ["name" .= ("fakeNewUser" :: Text)]
+        err <- asInwardErrorUnsafe <$> inwardBrigCall "i/users" (encode o)
+        expectErr IForbiddenEndpoint err
+
+    it "should only accept /federation/ paths (also when there are .. segments)" $
+      runTestFederator env $ do
+        let o = object ["name" .= ("fakeNewUser" :: Text)]
+        err <- asInwardErrorUnsafe <$> inwardBrigCall "federation/../i/users" (encode o)
+        expectErr IForbiddenEndpoint err
```

Comment:
dear reviewers, I'd love ideas on what other things to test. Would any property tests make sense here?

Reply:
I think having property tests for `sanitizePath` would be useful, but of course only insofar as the generator produces realistic URL inputs. As for ad-hoc test cases, you could maybe add some with non-ascii character (both percent-encoded and not), query string and/or fragments.

Reply:
I agree, some (unit) tests for `sanitizePath` would be good to have, including one that has "federation" as the only path segment.

Reply:
I added a bunch of unit tests, take a look.

Reply:
Nice! :+1: